File: Python/day1_puzzle2.py
import functools

with open("../input/day1.txt", "r") as f:
    list1, list2 = zip(*[line.strip().split("   ") for line in f.readlines()])

list1 = list(map(int, list1))
list2 = list(map(int, list2))

result = functools.reduce(lambda s, i: s + i * list2.count(i), list1, 0)
# result = sum(i * list2.count(i) for i in list1)

print(result)

File: javascript/day1_puzzle2.js
const fs = require('fs');

const list1 = [];
const list2 = [];

fs.readFileSync("../_input/day1.txt", "utf8").split("\n").forEach(line => {
  if (!line) return;
  const [a, b] = line.split("   ");
  list1.push(parseInt(a));
  list2.push(parseInt(b));
});

result = list1.reduce((sum, i) => {
  return sum + i * list2.filter(x => x === i).length;
}, 0);

console.log(result);
